code/src/components/app-button/AppButton.scss

```diff
@@ -1,40 +1,142 @@
 @import "@design-system";
 
-// @TODO: It is just an example. Remove it or modify later
-.button__contained {
-  // Example of using variable from design system
-  background-color: var(--spa-button-bg-color, $purple);
+@mixin button-size($padding, $font-size) {
+  padding: spacing(2) spacing($padding);
+  font-size: $font-size;
+}
+
+@mixin focus-style($color) {
+  &:focus {
+    outline: 2px solid $color;
+    outline-offset: 2px;
+  }
+}
+
+@mixin hover-effect($bg-color, $text-color, $border-color: null) {
+  &:hover {
+    background-color: var(--spa-button-hover-bg-color, $bg-color);
+    color: var(--spa-button-hover-text-color, $text-color);
+    @if $border-color {
+      border-color: var(--spa-button-hover-border-color, $border-color);
+    }
```

Comment:
Maybe we should also pass variables from outside?

Reply:
yeah, I forgot about it. Thank you )